CMakeLists.txt

```diff
@@ -226,6 +226,13 @@ mark_as_advanced(CURL_DISABLE_TFTP)
 option(CURL_DISABLE_VERBOSE_STRINGS "disables verbose strings" OFF)
 mark_as_advanced(CURL_DISABLE_VERBOSE_STRINGS)
 
+option(ENABLE_LIBPROXY "Define if you want to enable libproxy support" ON)
+if(ENABLE_LIBPROXY)
+  find_package(libproxy REQUIRED)
+  set(ENABLE_LIBPROXY ON)
+  list(APPEND CURL_LIBS ${LIBPROXY_LIBRARIES})
+endif()
+
```

Comment:
Thanks for your feedback, i will updated it.

Reply:
Actually the default value has been set on purpose. I want  to ensure that newer builds of curl are using it by default. I could change the naming then to  disable libproxy. Not sure how to deal with it here.

Reply:
By default if available - this is what all other options are doing (unless they're OFF by default).
I don't want my scripts suddenly fail because of a new optional component.

Reply:
> By default if available - this is what all other options are doing (unless they're OFF by default).
> I don't want my scripts suddenly fail because of a new optional component.

If you do "by default if available" then it's going to be enabled at random depending on what build dependencies are specified and how eagle-eyed the human packager is to check all the various build options. [Here's the standard "don't do that"](https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Automagic_dependencies) wiki page. Humans are just not smart enough to avoid traps like this. Please pick a default and force people building curl to deviate intentionally.

libproxy should definitely be enabled by default on Linux platforms. Not sure about others.

P.S. If you use meson instead of CMake, it is possible to safely use auto dependencies. But doesn't matter, because you use CMake.

Reply:
Ah, I'll note I disagree with the wiki page here:

> In the case of binary-based distributions, like RPM or DEB based ones, automagic dependencies does not change anything: if the user has something installed and is building by hand, it's usually what he wants to enable, while if it's the maintainer, he'll just have to add a dependency over the packages required to run the binaries he has created.

Basically the wiki page suggests this is only bad for Gentoo and not normal Linux distributions. It's not true. If you use automagic dependencies, Fedora/Debian/Ubuntu/yocto/buildroot/whatever packagers probably won't notice, and whether the feature gets enabled or disabled is likely to depend on what BuildRequires are specified. In short, not a good idea. Failing the build unless you pass `-DENABLE_FOO=OFF` is a cheap way to avoid broken downstream packages and hardly difficult to update your build scripts to handle.

Reply:
I'm generally in the "OFF by default" camp for everything, but this is an established pattern in curl and the only thing worse than this pattern, IMHO, is inconsistency.

Reply:
I think the general curl pattern is that we go with features OFF to start with, then over time if we deem the "thing" almost a defacto thing that most users want and use, we can *possibly* switch the default to ON if present and working.

Reply:
You partially inspired me to write a blog post. [See rule three](https://blogs.gnome.org/mcatanzaro/2022/07/15/best-practices-for-build-options/).